### ISSUE TITLE
[next-master] glitches in mirrored pairs and open/close context

### ISSUE THREAD

**Issue by arrowtype** [cjdunn/MM2SpaceCenter#26]
This is an issue I’m filing to track some debugging in a logical place. This issue is basically copying in the conversation from https://github.com/cjdunn/MM2SpaceCenter/pull/18.

## The issue

https://github.com/cjdunn/MM2SpaceCenter/pull/22 was made to correct the problem of a mirrored pair not showing up if "show open+close context" was active. It helps, but also has issues.

Now, in the `next-master` branch, there are unexpected interactions of "show mirrored pair" and "show open+close context".

As of (https://github.com/cjdunn/MM2SpaceCenter/pull/22), I can get at least three states:

1. **Correct:** One mirrored pair, for pairs containing lowercase or figures, e.g. `e a`, `slash six.hiviz`, `bracketleft a`, or for pairs not containing punctuation, e.g. `A T` or `R e`
2. **Incorrect:** Doubled mirrored pairs, for pairs containing punctuation and uppercase, e.g. `bracketleft T`, or uppercase and figure, e.g. `R 6`.
3. **Confusing edge case:** A strange case in which `slash O.RECT` breaks the space center string entirely.

<img width="959" alt="image" src="https://user-images.githubusercontent.com/45946693/158861984-2eaed851-35c1-4493-9109-bbf5d0790576.png">

I can literally only trigger the last case with `slash O.RECT`. The font I’m trying this on has other similar suffixes, like R.RECT and Q.RECT, but those don’t cause the issue. It’s weird.

My hypothesis/hope is: we might be able to solve all three problems with a bit of refactoring.

Probably, the script can be slightly refactored to eliminate a few if/else blocks that might be causing some of the issues.

We could potentially do something like:

```py
# set mirrored pair if option is check, else set as empty string
mirroredPair = self.pairMirrored(self.pair) if self.w.mirroredPair.get() else ""

# set open close context if option is check, else set as empty string
openCloseContext =  openCloseContextReturn(self.pair) if self.w.openCloseContext.get() else ""

# form the text, with options in front if they are set
text = f"{mirroredPair} {openCloseContext} {self.getSpacingString(pairstring)}"
```

...were we set partial strings based on settings, then make the `text` string only once, putting together the pieces. The actual code is more complex to handle different contexts, so maybe my hope is overly idealistic, but I’ll explore this route.

## Where is the best place show open+close context?

At one point, the open+close context was placed in front of the string, but since then, it was moved inside of the string.

Example: if the pair is `T bracketright`, there are two main options for how it can be shown:
- in front: `[T] HHT]HOT]OO`
- inside: `HH[T]HO[T]OO`

Advantages for having it in front: 
- When I look at it, I sometimes want to check: "is `T` in the middle of `H` and `]`?" If the string is `HH[T]H`, I can’t see that.
- Having it in front of the spacing string also makes it more obvious to users what effect the option has. 

Advantages for having it inside:
- For things like quotes, it is quite advantageous to see the context of a string like `nn‘s’no‘s’oo`, to show how the quotes are kerned against control characters.
- It is more compact.

I’m split. I suppose for quotes, the advantage of having it inside is extremely helpful, and that might be important enough to leave it inside. I won’t change it for now, but I’m open to opinions!

## What is the best way to show the mirrored pair?

I could go either way. It feels a little simpler to understand as just `LRL`, but I can imagine that it might be helpful to have `LRLR`, to see not only how R fits between L, but how L sits between R. Maybe that is redundant / unnecessary?

I’m unsure. I often just flip the pair (command F) to check the opposite mirroring. However, this could potentially disrupt one’s place in the pair list. So, I will also leave this as LRLR for now, but I’m somewhat uncertain that this is the best option.

**Comment by arrowtype** [comment]
This is mostly fixed by #28, so I’ll split the edge case problem into a new issue.
